### PR TITLE
feat: add grouping for exposing and removing

### PR DESCRIPTION
**Pull request by nichmor**
new output:
Screenshot 2024-10-30 at 15 31 57
Screenshot 2024-10-30 at 15 34 04

### CODE REVIEW

File: src/global/project/manifest.rs [hunk 1]
```diff
@@ -357,11 +357,11 @@ impl Manifest {
...
 
     /// Removes exposed mapping from the manifest
-    pub fn remove_exposed_name(
+    pub fn remove_exposed_name<'a>(
         &mut self,
         env_name: &EnvironmentName,
-        exposed_name: &ExposedName,
-    ) -> miette::Result<()> {
+        exposed_name: &'a ExposedName,
+    ) -> miette::Result<&'a ExposedName> {
```
Comment (Hofer-Julian) [inline src/global/project/manifest.rs:364]: I don't get that one. Why would you need to return the same exposed_name that you passed yourself to the function?

**Review by Hofer-Julian — COMMENTED** [review]
Works as expected, great job!

I only had one comment.
After CI is green, we can merge this 🚀

File: src/global/project/manifest.rs (hunk 1 above)
Comment (nichmor) [reply src/global/project/manifest.rs]: removed it

**Comment by nichmor** [comment]
> Works as expected, great job!
> 
> I only had one comment. After CI is green, we can merge this 🚀

fixed the test. should be green now

**Comment by Hofer-Julian** [comment]
> > Works as expected, great job!
> > I only had one comment. After CI is green, we can merge this 🚀
> 
> fixed the test. should be green now

There's still a lint error :)

**Review by Hofer-Julian — APPROVED** [review]
Awesome work!

**Comment by ruben-arts** [comment]
```
❯ pixi g install python=3.11 -e p
Global environments as specified in '/home/rarts/.pixi/manifests/pixi-global.toml'
└── p (installed)
    ├─ dependencies: python 3.11.10
    └─ exposes: 2to3, 2to3-3.11, idle3, idle3.11, pydoc, pydoc3, pydoc3.11, python, python3, python3-config, python3.1, python3.11, python3.11-config

❯ pixi g add matplotlib -e p
✔ Updated package matplotlib 3.9.2 in environment p.
✔ Updated executable 2to3 of environment p.
✔ Updated executable python3 of environment p.
✔ Updated executable python3.11 of environment p.
✔ Updated executable python3.11-config of environment p.
✔ Updated executable python3-config of environment p.
✔ Updated executable pydoc of environment p.
✔ Updated executable pydoc3 of environment p.
✔ Updated executable pydoc3.11 of environment p.
✔ Updated executable idle3.11 of environment p.
✔ Updated executable python of environment p.
✔ Updated executable python3.1 of environment p.
✔ Updated executable idle3 of environment p.
✔ Updated executable 2to3-3.11 of environment p.
✔ Added package matplotlib=3.9.2 to environment p.

❯ pixi g rm python -e p
✔ Updated package python 3.11.10 -> 3.12.7 in environment p.
✔ Removed exposed executables from environment p:
   - 2to3
   - 2to3-3.11
   - idle3
   - idle3.11
   - pydoc
   - pydoc3
   - pydoc3.11
   - python
   - python3
   - python3-config
   - python3.1
   - python3.11
   - python3.11-config
```
I think `Updated` should get the same output. Also I think there is a bug in the update exe logic but this is not added in this PR. As It should not have told me it updated those executables in this case.

**Pull request merged**